src/test/net_tests.cpp

```diff
@@ -771,4 +773,102 @@ BOOST_AUTO_TEST_CASE(PoissonNextSend)
     g_mock_deterministic_tests = false;
 }
 
+std::vector<NodeEvictionCandidate> GetRandomNodeEvictionCandidates(const int n_candidates, FastRandomContext& random_context)
+{
+    std::vector<NodeEvictionCandidate> candidates;
+    for (int id = 0; id < n_candidates; ++id) {
+        candidates.push_back({/* id */ id,
+                              /* nTimeConnected */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* nMinPingUsecTime */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* nLastBlockTime */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* nLastTXTime */ static_cast<int64_t>(random_context.randrange(100)),
+                              /* fRelevantServices */ random_context.randbool(),
+                              /* fRelayTxes */ random_context.randbool(),
+                              /* fBloomFilter */ random_context.randbool(),
+                              /* nKeyedNetGroup */ random_context.randrange(100),
+                              /* prefer_evict */ random_context.randbool(),
+                              /* m_is_local */ random_context.randbool()});
+    }
+    return candidates;
+}
+
+// Returns true if any of the node ids in node_ids are selected for eviction.
+bool IsEvicted(std::vector<NodeEvictionCandidate> candidates, const std::vector<NodeId> node_ids, FastRandomContext& random_context)
+{
+    Shuffle(candidates.begin(), candidates.end(), random_context);
+    const Optional<NodeId> evicted_node_id = SelectNodeToEvict(std::move(candidates));
+    if (!evicted_node_id) {
+        return false;
+    }
+    return std::find(node_ids.begin(), node_ids.end(), *evicted_node_id) != node_ids.end();
+}
+
+// Create number_of_nodes random nodes, apply setup function candidate_setup_fn,
+// apply eviction logic and then return true if any of the node ids in node_ids
+// are selected for eviction.
+bool IsEvicted(const int number_of_nodes, std::function<void(NodeEvictionCandidate&)> candidate_setup_fn, const std::vector<NodeId> node_ids, FastRandomContext& random_context)
+{
+    std::vector<NodeEvictionCandidate> candidates = GetRandomNodeEvictionCandidates(number_of_nodes, random_context);
+    for (size_t i = 0; i < candidates.size(); ++i) {
```

Comment:
nit: could use ranged for loop?

Reply:
Oh, of course. A prior version used the index. Thanks!